app/src/main/java/com/vero/tasky/core/presentation/navigation/RootNavigation.kt

```diff
@@ -66,26 +72,75 @@ fun RootNavigation(
             route = Screens.Event.route + agendaItemScreenParameters,
             arguments = agendaItemArgumentTypes
         ) {
+
             EventDetailsScreen(
-                navigateBack = { navController.popBackStack() }
+                navigateBack = { navController.popBackStack() },
+                onEditTitle = { title ->
+                    navController.navigate(
+                        Screens.EditTitle.route + "?$EDIT_TITLE_VALUE=$title"
+                    )
+                },
+                onEditDescription = { description ->
+                    navController.navigate(
+                        Screens.EditDescription.route + "?$EDIT_DESCRIPTION_VALUE=$description"
+                    )
+                },
+                title = it.savedStateHandle.get(EDIT_TITLE_VALUE),
+                description = it.savedStateHandle.get(EDIT_DESCRIPTION_VALUE),
+            )
+        }
+        composable(
+            route = Screens.EditTitle.route + "?$EDIT_TITLE_VALUE={$EDIT_TITLE_VALUE}",
+            arguments = listOf(
+                navArgument(EDIT_TITLE_VALUE) {
+                    type = NavType.StringType
+                }
+            )
+        ) {
+            val text = it.arguments?.getString(EDIT_TITLE_VALUE) ?: ""
+            EditTextScreen(
+                label = R.string.edit_title,
+                textValue = text,
+                navBack = { navController.popBackStack() },
+                save = { resultStr ->
+                    navController.previousBackStackEntry
+                        ?.savedStateHandle
+                        ?.set(EDIT_TITLE_VALUE, resultStr)
+                    navController.popBackStack()
+                },
+                textStyle = MaterialTheme.typography.Inter400Size26
+            )
+        }
+        composable(
+            route = Screens.EditDescription.route + "?$EDIT_DESCRIPTION_VALUE={$EDIT_DESCRIPTION_VALUE}",
+            arguments = listOf(
+                navArgument(EDIT_DESCRIPTION_VALUE) {
+                    type = NavType.StringType
+                }
+            )
+        ) {
+            val text = it.arguments?.getString(EDIT_DESCRIPTION_VALUE) ?: ""
+            EditTextScreen(
+                label = R.string.edit_description,
+                textValue = text,
+                navBack = { navController.popBackStack() },
+                save = { resultStr ->
+                    navController.previousBackStackEntry
+                        ?.savedStateHandle
```

Comment:
You are creating 2 routes. One for title and one for description

You could just have 1 route, with 2 optional query parameters 

So if the query parameter "title" is set, then you edit the title. Else, it's the description